File: C/src/cap2/num1.c
/* cd ~/Documentos/ProjetosUbuntu/C/ 
nano src/prog1.c  (ou nano | prog1.c |  em caso de omissão, nome é solicitado com Ctrl+O)
gcc src/prog1.c | -o bin/prog1 |   em caso de omissão, a.out é gerado automaticamente em ~/Documentos/ProjetosUbuntu/C/
./bin/prog1 */

// Programa: Manipulacao de Números - Capitulo 2 (C)
// Ambiente: Ubuntu - gcc / nano

#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]){
    const int N1 = 9;
    int n1;
    
    printf("LITERAL NUMERICA:\n");
    printf("%d\n", 41);
        
    printf("\n%s", "DECLARACAO DE VARIAVEL -> VALOR INDEFINIDO:\n");
    printf("%s%d\n", "n1 = ", n1);
     
    n1 = 7;
    printf("\nINICIALIZACAO:\n");
    printf("n1 = %d\n", n1);
    
    n1 = 10;
    printf("\nNOVA ATRIBUICAO:\n");
    printf("n1 = %d\n", n1);
    
    printf("\nENTRADA VIA ARGUMENTO:\n");
    printf("n1 = %d\n", (argc > 1) ? atoi(argv[1]) : 0);


    printf("\nENTRADA INTERATIVA:\n");
    printf("Digite um valor: ");
    scanf("%d", &n1);
    printf("n1 = %d\n", n1);
    
    printf("\nOPERACOES ARITMETICAS:\n");
    printf("n1 + N1 = %d\n", n1 + N1);
    printf("n1 - N1 = %d\n", n1 - N1);
    printf("n1 * N1 = %d\n", n1 * N1);
    printf("n1 / N1 = %d\n", n1 / N1);
    printf("n1 %% N1 = %d\n", n1 % N1);
    printf("\n");   	
    return 0;
}

File: Python/num1.py
#!/usr/bin/env python3
# Programa: Manipulacao de Números - Capitulo 2 (Python)
# Ambiente: Ubuntu / vscode
import sys, random
argv1 = int(sys.argv[1]) if len(sys.argv) > 1 and sys.argv[1].isdigit() else None
# ATENÇÃO: número negativo e/ou decimal informado em linha de comando é convertido a None
N1 = random.randint(1,9)
n1 = None

print("LITERAL NUMERICA:")
print("\033[7m41\033[0m")
 
print("\nVALOR INICIAL DA VARIAVEL:")
print("n1 = ", n1)
 
n1 = 7
print("\nNOVA ATRIBUICAO:")
print("n1 = ", n1)
 
print("\nENTRADA VIA ARGUMENTO:")
n1 = argv1
print("n1 = ", n1)

print("\nENTRADA INTERATIVA:")
x = input("Digite um valor inteiro nao negativo: ")     # número negativo e/ou decimal é convertido a 0
n1 = int(x) if x.isdigit() else 0
print("n1 = ", n1)
  
print("\nOPERACOES ARITMETICAS:")
print("n1 + N1 = ", n1 + N1)
print("n1 - N1 = ", n1 - N1)
print("n1 * N1 = ", n1 * N1)
print("n1 / N1 = ", n1 / N1)
print("n1 // N1 = ", n1 // N1)
print("n1 % N1 = ", n1 % N1)
print("n1 ** N1 = ", n1 ** N1)
  	
print("\nDESAFIO:")
x = input("Adivinhe o valor da constante N1 (1-9): ")
palpite = int(x) if x.isdigit() else 0
print(f"Parabéns, você acertou! Era {N1}") if palpite == N1 else print(f"Não foi desta vez! Era {N1}")
